src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -313,13 +313,35 @@ protected override void Dispose(bool disposing)
 
             if (disposing && !IsCompleted)
             {
-                _connector.CloseOngoingOperations();
+                _connector.CloseOngoingOperations(async: false).GetAwaiter().GetResult();
                 Rollback();
             }
 
             IsDisposed = true;
         }
 
+#if !NET461 && !NETSTANDARD2_0
+        /// <summary>
+        /// Disposes the transaction, rolling it back if it is still pending.
+        /// </summary>
+        public override async ValueTask DisposeAsync()
+        {
+            if (IsDisposed)
+                return;
+
+            if (!IsCompleted)
+            {
+                using (NoSynchronizationContextScope.Enter())
+                {
+                    await _connector.CloseOngoingOperations(async: true);
+                    await Rollback(async: true);
+                }
+            }
```

Comment:
No need in braces inside `if`.

Reply:
As in https://github.com/npgsql/npgsql/pull/2599#discussion_r319973163, for multiline cases I'd rather put the braces in to avoid any ambiguity. Maybe EF Core is having a bad influence, but the bad indentation in that comment's code shows that sometimes it makes sense...

BTW I also prefer braces when the if condition itself is multiline, e.g. https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/blob/210f94ceb42006997d0000610f567988dc510627/src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs#L154

Reply:
Up to you. I just don't like double closing brackets in the such cases which also can be seen in `foreach-if` cases and so on. Maybe `using var scope = NoSynchronizationContextScope.Enter();`?